apps/src/studio/blocks.js

```diff
@@ -1809,6 +1809,45 @@ exports.install = function (blockly, blockInstallOptions) {
     });
   };
 
+  blockly.Blocks.studio_whenSpriteAndGroupCollide = {
+    // Block to handle event when the Left arrow button is pressed.
+    helpUrl: '',
+    init: function () {
+      this.setHSV(140, 1.00, 0.74);
+
+      var dropdown1 = spriteNumberTextDropdown(msg.whenSpriteN);
+      var dropdown2 = createSpriteGroupDropdown(
+          msg.collidesWithAnySpriteName,
+          value => endLabel.setText(msg.toTouchedSpriteName(
+              {spriteName: stripQuotes(value)})));
+      this.appendDummyInput()
+          .appendTitle(dropdown1, 'SPRITE')
+          .appendTitle(dropdown2, 'SPRITENAME');
+      this.appendDummyInput();
+      this.appendValueInput('GROUPMEMBER')
+          .setInline(true)
+          .appendTitle(msg.set());
+      var endLabel = new Blockly.FieldLabel(msg.toTouchedSpriteName(
+              {spriteName: stripQuotes(dropdown2.getValue())}));
+      this.appendDummyInput()
+          .setInline(true)
+          .appendTitle(endLabel);
+
+      this.setPreviousStatement(false);
+      this.setNextStatement(true);
+      this.setTooltip(msg.whenSpriteAndGroupCollideTooltip());
+    },
+  };
+
+  generator.studio_whenSpriteAndGroupCollide = function () {
+    var varName = Blockly.JavaScript.valueToCode(this, 'GROUPMEMBER',
+        Blockly.JavaScript.ORDER_NONE);
+    // Sprite index vars need to be 1-indexed, but the callback arg will be
+    // 0-indexed, so add 1.
+    return `${varName} = extraArgs[0] + 1;\n`;
```

Comment:
Also, just pushed a commit to use named parameters like `touchedSpriteIndex` instead of `extraArgs[0]`. PTAL, but it's mostly in the event handling code that Josh should look at anyway.